plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codemodernizer/model/CodeModernizerArtifact.kt

```diff
@@ -99,7 +100,17 @@ open class CodeModernizerArtifact(
             if (!patchesDir.isDirectory()) {
                 throw RuntimeException("Expected root for patches was not a directory.")
             }
-            return patchesDir.walk()
+            val patchPathSequence = patchesDir.walk()
+            val patchFileList = patchPathSequence.map { fileSystem.findFileByNioFile(it) }.toList()
+            if (patchFileList[0] == null) {
+                ApplicationManager.getApplication().invokeAndWait {
+                    ApplicationManager.getApplication().runWriteAction {
```

Comment:
Would like feedback if anyone has a better understanding of the necessity of `invokeAndWait` plus `runWriteAction`. Seems like it comes down to thread safety

Reply:
This seems to summarize my concerns: https://javarevisited.blogspot.com/2011/09/invokeandwait-invokelater-swing-example.html#axzz8Tjdw80jd